R/CompDbSource.R

```diff
@@ -162,56 +162,128 @@ MassBankSource <- function(release = "2021.03", ...) {
+#'
+#' TODO: document PARAMETERS
 #' @author Nir Shahaf, Johannes Rainer
 #'
 #' @examples
+#'
+#' \dontrun{
+#' ## Load a test data
+#' load(system.file("extdata", "sps_test.Rdata", package = "MetaboAnnotation"))
+#' ## We need to define the adduct annotation(s) for each spectrum. Here we
+#' ## assume all MS2 spectra to represent [M-H]- ions.
+#' sps_test$precursorAdduct <- "[M-H]-"
+#'
+#' ## Match the spectra against MetFrag
+#' res <- matchSpectra(sps_test, MetFragSource(), MetFragParam())
+#' res
+#'
+#' ## By default a *match* is reported even if for a query spectrum the call
+#' ## failed (either because of a server timeout or because the query spectrum
+#' ## had to few fragment peaks. For each query the status message from the
+#' ## server is reported (with column "server_status"). The status for valid
+#' ## matches from the server is "SUCCESS" while NA is reported for spectra
+#' ## with too few peaks.
+#' table(res$server_status, useNA = "ifany")
+#'
+#' ## The `filterMatches` function can be used to clean up the results and
+#' ## remove any such non-matches.
+#' idx <- which(res$server_status == "SUCCESS")
+#' res <- filterMatches(res, SelectMatchesParam(index = idx))
+#' res
+#' }
+NULL
 
-#' We Assume that the user has added the precursor adduct annotation(s) to the spectra data,
-#' 	under the column 'precursorAdduct' e.g.:
-#' load(system.file("extdata", "sps_test.Rdata",package = "MetaboAnnotation"))
-#' spectraData(sps_test)$precursorAdduct = rep("[M-H]-", nrow = length(sps_test))
-#' Define MetFrag parameters object and make connection with the online server:
-#' mfParam = MetFragParam()
-#' mfSrc = MetFragSource()
-#' 
-#' testMatch <- matchSpectra(sps_test,mfSrc,MetFragParam())
+#' @rdname MetFrag
+#'
+#' @export
 setClass("MetFragParam",
          slots = c(
-		 #  validAdducts
-			validAdducts = "list",
-		 # database parameters -> how to retrieve candidates
-			metfragdatabasetype = "character",
-			OfflineSpectralDatabaseFile = "character",
-		 # peak matching parameters	
-			fragmentpeakmatchabsolutemassdeviation = "numeric",
-			fragmentpeakmatchrelativemassdeviation = "numeric",
-			databasesearchrelativemassdeviation = "numeric",
-		 # scoring parameters
-			MetFragScoreTypes = "character",
-			MetFragScoreWeights = "numeric",
-		 # default class filtering params 	
+             validAdducts = "list",
+             ## database parameters -> how to retrieve candidates
+             metfragdatabasetype = "character",
+             OfflineSpectralDatabaseFile = "character",
+             ## peak matching parameters
+             fragmentpeakmatchabsolutemassdeviation = "numeric",
+             fragmentpeakmatchrelativemassdeviation = "numeric",
+             databasesearchrelativemassdeviation = "numeric",
+             ## scoring parameters
+             MetFragScoreTypes = "character",
+             MetFragScoreWeights = "numeric",
+             ## default class filtering params
 			 requirePrecursor = "logical",
```

Comment:
do we need these parameters at all? `requirePrecursor`, `requirePrecursorPeak`, `toleranceRt`, `percentRt`?

Reply:
yes, indeed: RT information is stored on the server and will be used to increase accuracy (also with non-identical chromatographic methods) and 'requirePrecursor' set to false can perhaps be used for similarity searches of chemically related ions.